src/pcapi/routes/native/v1/bookings.py

```diff
@@ -0,0 +1,108 @@
+from datetime import datetime
+
+from sqlalchemy.orm import Load
+from sqlalchemy.orm import joinedload
+
+import pcapi.core.bookings.api as bookings_api
+import pcapi.core.bookings.exceptions as exceptions
+from pcapi.core.bookings.models import Booking
+from pcapi.core.offerers.models import Venue
+from pcapi.core.offers.models import Offer
+from pcapi.core.offers.models import Stock
+from pcapi.core.users.models import User
+from pcapi.models.api_errors import ApiErrors
+from pcapi.routes.native.security import authenticated_user_required
+from pcapi.routes.native.v1.serialization.bookings import BookOfferRequest
+from pcapi.routes.native.v1.serialization.bookings import BookingReponse
+from pcapi.routes.native.v1.serialization.bookings import BookingsResponse
+from pcapi.serialization.decorator import spectree_serialize
+
+from . import blueprint
+
+
+@blueprint.native_v1.route("/book_offer", methods=["POST"])
+@spectree_serialize(api=blueprint.api, on_success_status=204)
+@authenticated_user_required
+def book_offer(user: User, body: BookOfferRequest) -> None:
+    stock = Stock.query.get(body.stock_id)
+    if not stock:
+        raise ApiErrors({"stock": "stock introuvable"}, status_code=404)
```

Comment:
```suggestion
        raise ApiErrors({"stock": "stock introuvable"}, status_code=400)
```

`/book_offer` existe bien donc ce ne devrait pas être un 404 (je sais, on a cette erreur ici et là, il faudra que je prenne un peu de temps pour corriger)

Reply:
ok je connaissais pas cette logique : on renvoie une 404 si la route est introuvable, par exemple si stock_id était écrit dans l'url... ?

Reply:
Si le `stock_id` était dans l'url, le 404 serait tout à fait adapté:
La définition du 404:

> The server has not found anything matching the Request-URI.